libminifi/src/utils/ThreadPool.cpp

```diff
@@ -64,8 +64,10 @@ void ThreadPool<T>::run_tasks(std::shared_ptr<WorkerThread> thread) {
         }
       }
     } else {
-      std::unique_lock<std::mutex> lock(worker_queue_mutex_);
-      tasks_available_.wait(lock);
+      // This means that the threadpool is running, but the ConcurrentQueue is stopped -> shouldn't happen
+      if (running_.load()) {
+        std::this_thread::sleep_for(std::chrono::milliseconds(0));
```

Comment:
If this shouldn't happen, how sleeping for 0 milliseconds will help?

Reply:
Meant to sleep for 1 msec, corrected that. 

In case this somehow occurs in a transient state (during startup or shutdown), this prevent the worker threads to busy wait (dequeue fails immediately and the workers would retry without sleep).